src/helpers/MethodInvocationHelpers.ts

```diff
@@ -91,49 +103,134 @@ export const autoSelectAccountForScope = (
+  const address = selectedAccount.split(':')[2] ?? '';
+  const example = await generateSolanaMethodExamples(method, address);
+
+  if (!example) {
+    console.error(`❌ No example found for Solana method: ${method}`);
+    return null;
+  }
+
+  return example;
+};
+
+/**
+ * Handles EVM-specific method preparation.
  *
  * @param method - The method name to invoke.
- * @param caipChainId - The CAIP chain ID.
  * @param selectedAccount - The selected account for this scope.
+ * @param caipChainId - The CAIP chain ID.
  * @param metamaskOpenrpcDocument - The MetaMask OpenRPC document.
- * @param injectParams - Function to inject parameters for specific methods.
- * @param openRPCExampleToJSON - Function to convert OpenRPC examples to JSON.
- * @param METHODS_REQUIRING_PARAM_INJECTION - Object containing methods that require parameter injection.
  * @returns The prepared request object or null if method not found.
  */
-export const prepareMethodRequest = (
+const handleEVMMethod = async (
   method: string,
+  selectedAccount: CaipAccountId,
   caipChainId: CaipChainId,
-  selectedAccount: CaipAccountId | null,
   metamaskOpenrpcDocument: any,
-  injectParams: (
-    method: string,
-    params: Json,
-    account: CaipAccountId,
-    scope: CaipChainId,
-  ) => Json,
-  openRPCExampleToJSON: (methodObj: MethodObject) => Json,
-  METHODS_REQUIRING_PARAM_INJECTION: Record<string, boolean>,
-): Json | null => {
+): Promise<Json | null> => {
   const example = metamaskOpenrpcDocument?.methods.find(
     (methodObj: MethodObject) => methodObj.name === method,
   );
 
   if (!example) {
-    console.error(`❌ No example found for method: ${method}`);
+    console.error(`❌ No example found for EVM method: ${method}`);
     return null;
   }
 
-  let exampleParams: Json = openRPCExampleToJSON(example as MethodObject);
+  let exampleParams = openRPCExampleToJSON(example as MethodObject);
 
-  if (method in METHODS_REQUIRING_PARAM_INJECTION && selectedAccount) {
+  if (method in METHODS_REQUIRING_PARAM_INJECTION && exampleParams !== null) {
     exampleParams = injectParams(
       method,
       exampleParams,
       selectedAccount,
       caipChainId,
-    );
+    ) as { method: string; params: any };
+  }
+
+  return exampleParams;
+};
+
+/**
+ * Prepares a method request object for invocation.
+ *
+ * @param method - The method name to invoke.
+ * @param caipChainId - The CAIP chain ID.
+ * @param selectedAccount - The selected account for this scope.
+ * @param metamaskOpenrpcDocument - The MetaMask OpenRPC document.
+ * @returns The prepared request object or null if method not found.
+ */
+export const prepareMethodRequest = async (
+  method: string,
+  caipChainId: CaipChainId,
+  selectedAccount: CaipAccountId | null,
+  metamaskOpenrpcDocument: any,
+): Promise<Json | null> => {
+  const methodType = determineMethodType(caipChainId);
+
+  if (!selectedAccount) {
+    console.error(`❌ No account selected for method: ${method}`);
```

Comment:
Not sure if we want all these console errors since they will probably be occurring a lot and they won't necessarily help you identify how to proceed?

Reply:
Also looks like it can't reach here without a `selectedAccount` being defined

I'm not understanding the `handleDirectMethodInvoke` and how its only supposedly tied to e2e tests and not manually invoking methods?